src/lib/db.ts

```diff
@@ -519,6 +520,15 @@ export class DB {
     });
   }
 
+  public copyOtherDocument(document: DocumentModelType, newTitle?: string) {
+    const title = `${newTitle || document.title || this.stores.problem.title || "Untitled"}`;
```

Comment:
```suggestion
    const title = newTitle || `Copy of ${document.title}`;
```
The client should really provide the title; we don't need to go overboard with default options. OtherDocuments should all have titles, so this should be good enough.

Reply:
I'm already doing `Copy of ` in `document-workspace.tsx` which is why I removed it here. The reason I did `this.stores.problem.title` was because `document.title` only grabs Personal Documents or Learning Log Documents, not Problem Documents which required `this.stores.problem.title` to grab. Would this be a possible option to consider?

```suggestion
const { problem } = this.stores; 
const title = newTitle || documentTitle || problem.title;
```

Reply:
You've convinced me to go with the other option I considered -- make the title a required parameter and this function doesn't do any defaulting. All of the other options being discussed require that this otherwise generic function contain too much client-specific domain knowledge.

Reply:
Sounds good with me! It's now purely on the client side.